src/storage/mod.rs

```diff
@@ -19,22 +31,128 @@ pub trait Engine {
 /// table (like column names, column types, storage engine, ...). It's `access`
 /// method locks the table globally and returns a storage engine to access
 /// the table data.
-pub struct Table;
+
+pub enum DatabaseError{
+    Io(Error),
+    Bin(EncodingError),
+}
+
+impl From<io::Error> for DatabaseError {
+    fn from(err: io::Error) -> DatabaseError {
+        DatabaseError::Io(err)
+    }
+}
+
+impl  From<EncodingError> for DatabaseError {
+    fn from(err: EncodingError) -> DatabaseError {
+        DatabaseError::Bin(err)
+    }
+}
+
+/// A Enum for File Modes
+///
+/// Can be used to define the save and load configuration of open_file
+#[derive(Debug)]
+enum FileMode{LoadDefault, SaveDefault,}
+
+/// A Enum for Datatypes (will be removed later)
+#[repr(u8)]
+#[derive(Clone,Copy,Debug,RustcDecodable, RustcEncodable)]
+enum DataType{ Integer = 1, Float = 2,}
+impl DataType{
+    pub fn value(&self) -> u8{
+       *self as u8
+    }
+}
+
+pub struct Database {
+    name: String,
+}
+
+impl Database {
+    pub fn new_database(database_name: &str) -> Database{
+        Database{name: database_name.to_string()}
+    }
+    pub fn create_database(&self) -> Result<(),DatabaseError>{
+        println!("trying to create dir!");
+        try!(create_dir(&self.name));
+        println!("created dir");
+        Ok(())
+    }
+}
+
+#[derive(Debug,RustcDecodable, RustcEncodable)]
+pub struct Table {
+    engine_id: u8,
+    version_nmbr: u8,
+    magic_nmbr: u8,
```

Comment:
It doesn't really make sense the save the magic number in each table object. The magic number will never change and is just for file verification. So you could use a constant (`const MAGIC_NUMBER: u8 = 180`). Also: Usually magic numbers are 4 or 8 bytes wide.